Source/NonlinearSolvers/NewtonSolver.H

```diff
@@ -356,7 +357,8 @@ void NewtonSolver<Vec,Ops>::Solve ( Vec&         a_U,
        }
     }
 
-    if (!this->m_diagnostic_file.empty() && amrex::ParallelDescriptor::IOProcessor()) {
+    if (!this->m_diagnostic_file.empty() && amrex::ParallelDescriptor::IOProcessor() &&
+        a_step%this->m_diagnostic_interval==0) {
```

Comment:
You can make this change here to get the counting correct.
```suggestion
        (a_step + 1)%this->m_diagnostic_interval==0) {
```